airflow/models/baseoperator.py

```diff
@@ -1600,7 +1601,10 @@ def resume_execution(self, next_method: str, next_kwargs: dict[str, Any] | None,
             traceback = next_kwargs.get("traceback")
             if traceback is not None:
                 self.log.error("Trigger failed:\n%s", "\n".join(traceback))
-            raise TaskDeferralError(next_kwargs.get("error", "Unknown"))
+            if (error := next_kwargs.get("error", "Unknown")) == "Trigger timeout":
```

Comment:
I wonder if this can be made a bit more robust…

Reply:
@uranusjr any particular suggestion you have?